gooddata-sdk/gooddata_sdk/catalog/data_source/declarative_model/physical_model/table.py

```diff
@@ -0,0 +1,47 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any
+
+from gooddata_metadata_client.model.declarative_table import DeclarativeTable
+from gooddata_sdk.catalog.data_source.declarative_model.physical_model.column import CatalogDeclarativeColumn
+from gooddata_sdk.catalog.entity import CatalogTypeEntity
+
+
+class CatalogDeclarativeTable(CatalogTypeEntity):
+    def __init__(
+        self,
+        id: str,
+        type: str,
+        path: list[str],
+        columns: list[CatalogDeclarativeColumn],
+    ):
+        super(CatalogDeclarativeTable, self).__init__(id, type)
+        self.path = path
+        self.columns = columns
+
+    @classmethod
+    def from_api(cls, entity: dict[str, Any]) -> CatalogDeclarativeTable:
+        columns = [CatalogDeclarativeColumn.from_api(v) for v in entity["columns"]] if entity.get("columns") else []
+        return cls(
+            id=entity["id"],
+            type=entity["type"],
+            path=entity["path"],
+            columns=columns,
+        )
+
+    def to_api(self) -> DeclarativeTable:
+        kwargs: dict[str, Any] = {
+            "columns": [v.to_api() for v in self.columns],
+        }
```

Comment:
Do not put columns in kwargs. You can place columns directly in DeclarativeTable. Columns are mandatory.

Reply:
fixed